test/app/presenter/ChessBoardPresenter.spec.ts

```diff
@@ -74,3 +56,20 @@ function chessBoardViewMock(viewEventBus: ViewEventBus): ChessBoardView {
     hideAllAvailableMoves: jest.fn(),
   };
 }
+
+function chessboardStateMock(square: Square[], position: Position) {
+  const viewEvents: ViewEventBus = new InMemoryViewEventBus();
+  const view: ChessBoardView = chessBoardViewMock(viewEvents);
+  const model: ChessModel = {
+    squaresWithPiece: PIECES_START_POSITION,
+    move: jest.fn(),
+    possibleMoves(position: Square): Square[] {
+      return square;
+    },
+  };
+
+  new ChessBoardPresenter(view, model);
+
+  viewEvents.publish(new SquareWasClicked(position));
```

Comment:
To już powinno iść do konkretnego testu. Bo teraz pomieszales sekcje Given w When. ` viewEvents.publish(new SquareWasClicked(position));` to już when `when square A2 was clicked on the view` to nie moze byc czescia ustawiania mockow w given. To powinno byc napisane w tescie :) Inaczej kazdy test, ktory korzysta z ustawienia od razu by robil klik.

Reply:
A tak to OK :)

Reply:
aaaa no tak masz rację :P